iocage/lib/JailConfig.py

```diff
@@ -0,0 +1,612 @@
+import re
+
+import iocage.lib.JailConfigJSON
+import iocage.lib.JailConfigInterfaces
+import iocage.lib.JailConfigAddresses
+import iocage.lib.JailConfigResolver
+import iocage.lib.JailConfigFstab
+import iocage.lib.JailConfigLegacy
+import iocage.lib.JailConfigZFS
+import iocage.lib.helpers
+
+
+class JailConfig():
+
+    def __init__(self, data={}, jail=None, logger=None, new=False):
+
+        iocage.lib.helpers.init_logger(self, logger)
+
+        object.__setattr__(self, 'data', {})
+        object.__setattr__(self, 'special_properties', {})
+        object.__setattr__(self, 'legacy', False)
```

Comment:
`JailConfig. __setattr__` does special things, so we want to set this attributes with the native object implementation of `__setattr__`